types/types.go

```diff
@@ -23,67 +24,177 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+type State uint8
+
+const (
+	Unprocessed State = iota
+	Active
+	Silenced
+	Inhibited
+)
+
+// AlertStatus stores the state and values associated with an Alert.
+type AlertStatus struct {
+	status State
+	values []string
```

Comment:
Just having silence IDs here as a bunch of strings might fit with the UI currently offers, but it has semantically no real meaning from an API perspective.
What if we also want to show which alerts inhibited the given alert? How would we distinguish its ID from that of a silence.

I think the status should rather have "status", "silencedBy" (list of silence IDs), and "inhibitedBy" (list of alert IDs).

Also the attributes are both unexported so they shouldn't be marshalled by the JSON decoder.